cineast-api/src/main/java/org/vitrivr/cineast/api/messages/lookup/BooleanLookup.java

```diff
@@ -0,0 +1,139 @@
+package org.vitrivr.cineast.api.messages.lookup;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.commons.lang3.builder.ReflectionToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+import org.apache.commons.lang3.tuple.MutableTriple;
+import org.apache.commons.lang3.tuple.Triple;
+import org.graalvm.compiler.nodes.calc.ObjectEqualsNode;
+import org.vitrivr.cineast.api.messages.interfaces.Message;
+import org.vitrivr.cineast.api.messages.interfaces.MessageType;
+import org.vitrivr.cineast.core.data.providers.primitive.PrimitiveTypeProvider;
+import org.vitrivr.cineast.core.db.RelationalOperator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class BooleanLookup implements Message {
```

Comment:
Since this class seems to be at the core of the newly added functionality, it would be nice to have some documentation on this class